akka-persistence/src/main/scala/akka/persistence/PersistentActor.scala

```diff
@@ -406,7 +407,40 @@ abstract class UntypedPersistentActor extends UntypedActor with Eventsourced wit
 /**
  * Java API: an persistent actor - can be used to implement command or event sourcing.
  */
-abstract class AbstractPersistentActor extends AbstractActor with Eventsourced {
+abstract class AbstractPersistentActor extends AbstractActor with AbstractPersistentActorLike {
+  /**
+   * Recovery handler that receives persisted events during recovery. If a state snapshot
+   * has been captured and saved, this handler will receive a [[SnapshotOffer]] message
+   * followed by events that are younger than the offered snapshot.
+   *
+   * This handler must not have side-effects other than changing persistent actor state i.e. it
+   * should not perform actions that may fail, such as interacting with external services,
+   * for example.
+   *
+   * If there is a problem with recovering the state of the actor from the journal, the error
+   * will be logged and the actor will be stopped.
+   *
+   * @see [[Recovery]]
+   */
+  def createReceiveRecover(): AbstractActor.Receive
+
+  /**
+   * An persistent actor has to define its initial receive behavior by implementing
+   * the `createReceive` method, also known as the command handler. Typically
+   * validates commands against current state (and/or by communication with other actors).
+   * On successful validation, one or more events are derived from a command and
+   * these events are then persisted by calling `persist`.
+   */
+  def createReceive(): AbstractActor.Receive
+
+  // Note that abstract methods createReceiveRecover and createReceive are also defined in
+  // AbstractPersistentActorLike. They were included here also for binary compatibility reasons.
```

Comment:
when adding these two abstract methods here the diff for `javap -p` is identical before and after, except the order of some methods

Reply:
mima still complains though

Reply:
Hmmm, sounds good then.